tests/unit/test_placeholders_with_steps.py

```diff
@@ -214,6 +214,64 @@ def test_map_state_with_placeholders():
     result = Graph(workflow_definition).to_dict()
     assert result == expected_repr
 
+
+def test_map_state_with_placeholders():
+    map_item_value = MapItemValue(schema={
+        'name': str,
+        'age': str
+    })
+
+    map_item_index = MapItemIndex()
+
+    map_state = Map(
+        'MapState01',
+        parameters={
+            "MapIndex": map_item_index,
```

Comment:
question: is using multiple indexes valid? - seems we don't preclude the ability to do that, but would that generate invalid ASL? - does it make sense to protect users against it or is it overkill right now?

Reply:
Parameters can be anything you want. You can reference `$$.Map.Item.Index` as multiple times for different fields, but the value is always the same within the iteration.